apps/docs/content/guides/auth/auth-hooks/before-user-created-hook.mdx

```diff
@@ -0,0 +1,488 @@
+---
+id: 'before-user-created-hook'
+title: 'Before User Created Hook'
+subtitle: 'Prevent unwanted signups by inspecting and rejecting user creation requests'
+---
+
+This hook runs before a new user is created. It allows developers to inspect the incoming user object and optionally reject the request. Use this to enforce custom signup policies that Supabase Auth does not handle natively - such as blocking disposable email domains, restricting access by region or IP, or requiring that users belong to a specific email domain.
+
+The hook is triggered any time a new user would be created - including traditional email signups (signUp()), OAuth flows (signIn(\{ provider \})), SSO (signInWithSSO()), IdP-initiated SAML flows, and anonymous sign-ins. Before the user is inserted into the auth.users table, the payload is passed to your configured hook endpoint for evaluation.
+
+You can implement this hook using an HTTP endpoint or a Postgres function. If the hook returns an error object, the signup is denied and the user is not created. If the hook responds successfully (HTTP 200 or 204 with no error), the request proceeds as usual. This gives you full control over which users are allowed to register — and the flexibility to apply that logic server-side.
+
+**Inputs**
+
+Supabase Auth will send a payload containing these fields to your hook:
+
+| Field    | Type     | Description                                                                               |
+| -------- | -------- | ----------------------------------------------------------------------------------------- |
+| `header` | `object` | Metadata about the request. Includes IP address, request ID, and hook type.               |
+| `user`   | `object` | The user record that is about to be created. Matches the shape of the `auth.users` table. |
+
+<Tabs
+  scrollable
+  size="small"
+  type="underlined"
+>
+<TabPanel id="password-verification-attempt-json" label="JSON">
+
+```json
+{
+  "header": {
```

Comment:
Isn't this metadata? https://github.com/supabase/auth/blob/master/internal/hooks/v0hooks/v0hooks.go#L58

Reply:
Fixed